bin/check_hcid.py

```diff
@@ -0,0 +1,249 @@
+            except:
+                (
+                    percentage,
+                    num_clade_root,
+                    num_direct,
+                    ignore1,
+                    ignore2,
+                    raw_rank,
+                    ncbi,
+                    name,
+                ) = line.strip().split("\t")
+
+            ncbi = int(ncbi)
+            if ncbi not in taxid_map:
+                continue
+            percentage = float(percentage)
+            num_clade_root = int(num_clade_root)
+            num_direct = int(num_direct)
+            if num_direct > num_clade_root:
+                num_direct, num_clade_root = num_clade_root, num_direct
+            name = name.strip()
+            rank = raw_rank[0]
+
+            entries[ncbi] = {
+                "percentage": percentage,
+                "count": num_direct,
+                "count_descendants": num_clade_root,
+                "raw_rank": raw_rank,
+                "rank": rank,
+                "name": name,
+            }
+            counts[taxid_map[ncbi]]+=num_direct
+
+    return entries, counts
+
+def load_hcid_dict(hcid_file):
+    hcid_dict = {}
+    with open(hcid_file, "r") as f:
+        hcid = json.load(f)
+        for d in hcid:
+            if d["taxon_id"]:
+                hcid_dict[d["taxon_id"]] = d
+                hcid_dict[d["taxon_id"]]["classified_count"] = 0
+                hcid_dict[d["taxon_id"]]["classified_parent_count"] = 0
+                hcid_dict[d["taxon_id"]]["classified_found"] = False
+                hcid_dict[d["taxon_id"]]["classified_parent_found"] = False
+    return hcid_dict
+
+def check_report_for_hcid(hcid_dict, taxonomy_dir, kreport_file):
+    parents, children = load_from_taxonomy(taxonomy_dir)
+
+    taxid_map = {}
+    for d in hcid_dict:
+        if not d:
+            continue
+        taxids = set()
+        lookup = [d]
+        if "alt_taxon_ids" in hcid_dict:
+            lookup.extend(hcid_dict["alt_taxon_ids"])
+        while len(lookup) > 0:
+            child = lookup.pop()
+            if child in children:
+                lookup.extend(children[child])
+            taxid_map[child] = d
+        if d in parents:
+            taxid_map[parents[d]] = parents[d]
+
+    entries, counts = parse_report_file(kreport_file, taxid_map)
+    for taxid in hcid_dict:
+        hcid_dict[taxid]["classified_count"] = counts[taxid]
+        if taxid in parents and parents[taxid] in entries:
+            print(taxid, parents[taxid], parents[taxid] in taxid_map, parents[taxid] in entries)
+            hcid_dict[taxid]["classified_parent_count"]= entries[parents[taxid]]["count_descendants"]
+        if counts[taxid] > hcid_dict[taxid]["min_count"]:
+            hcid_dict[taxid]["classified_found"] = True
+        if counts[parents[taxid]] > hcid_dict[taxid]["min_count"]:
+            hcid_dict[taxid]["classified_parent_found"] = True
+
+
+def map_to_refs(query, reference):
```

Comment:
It might be an idea to grab refs programatically if they don't exist so we can just keep it running without having to manually curate them but that's hardly urgent

Reply:
I initially started downloading with entrez-direct using a script, but you often end up with random junk in the file too. Agree this would be a nice to have, but unless we think this list will be updated regularly I'm going to leave. If it comes up, I would vote for still having this file, but a separate utility script that could download and check references from a list of accessions and save